alkira/resource_alkira_connector_gcp_interconnect.go

```diff
@@ -0,0 +1,341 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraConnectorGcpInterconnect() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage GCP Interconnect.",
+
+		CreateContext: resourceConnectorGcpInterconnectCreate,
+		ReadContext:   resourceConnectorGcpInterconnectRead,
+		UpdateContext: resourceConnectorGcpInterconnectUpdate,
+		DeleteContext: resourceConnectorGcpInterconnectDelete,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, m interface{}) error {
+			client := m.(*alkira.AlkiraClient)
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the connector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"size": {
+				Description: "The size of the connector, one of `SMALL`, " +
+					"`MEDIUM`, `LARGE`, `2LARGE`, `5LARGE` or `10LARGE`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Description: "The description of the connector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "The CXP where the connector should be provisioned.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"enabled": {
+				Description: "Is the connector enabled. Default is `true`.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+			},
+			"group": {
+				Description: "The group of the connector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"tunnel_protocol": {
+				Description: "The tunnel protocol used by the connector." +
+					"Can be one of `GRE`, `IPSEC`, `VXLAN`, `VXLAN_GPE`.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"GRE", "IPSEC", "VXLAN", "VXLAN_GPE"}, false),
+			},
+			"billing_tag_ids": {
+				Description: "Billing tags to be associated with " +
+					"the resource. (see resource `alkira_billing_tag`).",
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+			},
+			"loopback_prefixes": {
```

Comment:
This could be confused with our `policy_prefix` resource.. Could we have an example of what the prefix looks like?